pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -707,6 +707,7 @@ func (c *Reconciler) createTaskRun(ctx context.Context, rprt *resources.Resolved
 		return c.PipelineClientSet.TektonV1beta1().TaskRuns(pr.Namespace).UpdateStatus(ctx, tr, metav1.UpdateOptions{})
 	}
 
+	rprt.PipelineTask = resources.ApplyPipelineTaskContexts(rprt.PipelineTask)
```

Comment:
I am happy to merge this as is, but setting the context variable here will only make it available to pipelineTask and will not be able to custom task. Since the `retries` is constant and does not change with the pipeline execution, we can set this while resolving pipeline task.